meta-resin-common/classes/image_types_resin.bbclass

```diff
@@ -255,6 +256,176 @@ IMAGE_CMD_resinos-img () {
     esac
 }
 
+
+#IMAGE_FEATURES_append = "${@bb.utils.contains('DISTRO_FEATURES', 'development-image', 'debug-tweaks', '', d)}"
+
+UBIMULTIVOL_BUILD = "boot rootfsA state data"
+UBINIZE_ARGS ?= "-m 0x4000 -p 0x200000"
+
+RESIN_BOOT_DIR ?= "${DEPLOY_DIR_IMAGE}/boot"
+RESIN_ROOTB_DIR ?= "${DEPLOY_DIR_IMAGE}/rootb"
+RESIN_STATE_DIR ?= "${DEPLOY_DIR_IMAGE}/state"
+RESIN_DATA_DIR ?= "${DEPLOY_DIR_IMAGE}/data_disk"
+
+IMAGE_CMD_resinos-ubi () {
+	MKUBIFS_ARGS_boot="-e 0x1f8000 -c 2000 -m 0x4000 -x lzo"
+	ADDITIONAL_MKUBIFS_ARGS_boot="-r ${RESIN_BOOT_DIR} -o ${DEPLOY_DIR_IMAGE}/boot.ubifs"
+	ADDITIONAL_UBINIZE_ARGS_boot="mode=ubi\nimage=${DEPLOY_DIR_IMAGE}/boot.ubifs\nvol_id=0\nvol_size=${RESIN_BOOT_SIZE}KiB\nvol_type=dynamic\nvol_name=${RESIN_BOOT_FS_LABEL}"
+
+	MKUBIFS_ARGS_rootfsA="-e 0x1f8000 -c 2000 -m 0x4000 -x lzo"
+	ADDITIONAL_MKUBIFS_ARGS_rootfsA="-r ${IMAGE_ROOTFS} -o ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfsA.ubifs"
+	# XXX dynamically set size, set below, after mkfs
+
+	MKUBIFS_ARGS_rootfsB="-e 0x1f8000 -c 2000 -m 0x4000 -x lzo"
+	ADDITIONAL_MKUBIFS_ARGS_rootfsB="-r ${RESIN_ROOTB_DIR} -o ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}.rootfsB.ubifs"
+	# XXX dynamically set size, set below, after mkfs
+
+	MKUBIFS_ARGS_state="-e 0x1f8000 -c 2000 -m 0x4000 -x lzo"
+	ADDITIONAL_MKUBIFS_ARGS_state="-r ${RESIN_STATE_DIR} -o ${DEPLOY_DIR_IMAGE}/state.ubifs"
+	ADDITIONAL_UBINIZE_ARGS_state="mode=ubi\nimage=${DEPLOY_DIR_IMAGE}/state.ubifs\nvol_id=3\nvol_size=40MiB\nvol_type=dynamic\nvol_name=${RESIN_STATE_FS_LABEL}"
+	# XXX resin-state partition size does not match resinos-img
+
+	MKUBIFS_ARGS_data="-e 0x1f8000 -c 10000 -m 0x4000 -x lzo"
+	ADDITIONAL_MKUBIFS_ARGS_data="-r ${DEPLOY_DIR_IMAGE}/data_disk -o ${DEPLOY_DIR_IMAGE}/data.ubifs"
+	ADDITIONAL_UBINIZE_ARGS_data="mode=ubi\nimage=${DEPLOY_DIR_IMAGE}/data.ubifs\nvol_id=4\nvol_type=dynamic\nvol_name=${RESIN_DATA_FS_LABEL}\nvol_flags=autoresize"
+
+	echo "Checking for Existing DIRs"
+	rm -rf "${RESIN_BOOT_DIR}/"
+	rm -rf "${RESIN_STATE_DIR}/"
+	rm -rf "${RESIN_ROOTB_DIR}/"
+	rm -rf "${DEPLOY_DIR_IMAGE}/images/"
+	rm -rf "${RESIN_DATA_DIR}/"
+
+	echo "Create Resin state DIR"
+	mkdir -p ${RESIN_STATE_DIR}
+	touch ${RESIN_STATE_DIR}/README_-_DO_NOT_DELETE_FILES_IN_THIS_DIRECTORY.txt
+
+	echo "Create Resin data DIR"
+	mkdir -p ${RESIN_DATA_DIR}
+
+	echo "Create Resin rootb DIR"
+	mkdir -p ${RESIN_ROOTB_DIR}
+
+	echo "Copying files in RESIN_BOOT_PARTITION_FILE"
+	mkdir ${RESIN_BOOT_DIR}
+	echo -n '' > ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+	for RESIN_BOOT_PARTITION_FILE in ${RESIN_BOOT_PARTITION_FILES}; do
```

Comment:
Is there anything else in our codebase that would benefit from such a function? Why would we wish to have fat as output? Would this code be immediately useful?